[dominopizzaaaa] iP

dominopizzaaaa opened the pull request:
# Teaching you about github
- Uses git
- Can be used for free
- Stores code
1. ~~Fast~~ __SUPER FAST__
2. Efficient
3. Free


A quote from myself...
> If you don't use github, its your loss

```
# Example of git commands
git add
git commit
git push
```
Use `git status` to list all new or modified files that haven't yet been committed.
- [ ] Watched tutorial
- [ ] Tried github out
- [ ] Created your first program! :+1:
> [!IMPORTANT]
Need to learn github to pass CS2103T

This site was built using [GitHub Pages](https://pages.github.com/).

Reminder: **All these are a _MUST_ do**

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

dominopizzaaaa committed:
Renamed Papadom.java and Changed its contents

dominopizzaaaa committed:
Made system able to take inputs, exit upon bye command

dominopizzaaaa committed:
Allow users to add to a list and display the list

dominopizzaaaa committed:
Used array for storing task instead for easy access and tracking

dominopizzaaaa committed:
created new class Task and able to mark and unmark tasks

dominopizzaaaa committed:
Added Javadocs

dominopizzaaaa committed:
added 3 new sub types of Task

dominopizzaaaa committed:
Changed array to arraylist, added functionalities for the 3 types

dominopizzaaaa committed:
done testing and adjustment to Papadom file

dominopizzaaaa committed:
Added exception handling

dominopizzaaaa committed:
Encapsulate dashes into a function

dominopizzaaaa committed:
Added delete function

dominopizzaaaa committed:
Added more exception handling

dominopizzaaaa committed:
Added enum

dominopizzaaaa committed:
Follow Java Conventions, added Javadocs

dominopizzaaaa committed:
Changing file structure

dominopizzaaaa committed:
Save tasks into tasks.txt

dominopizzaaaa committed:
Merge branch 'branch-Level-7'

dominopizzaaaa committed:
Store deadline dates as DateTime or Date

dominopizzaaaa committed:
Add new test cases

dominopizzaaaa committed:
Merge branch 'branch-Level-8'

dominopizzaaaa committed:
Add Parser, Storage and Ui classes

dominopizzaaaa committed:
Extract classes for more OOP

dominopizzaaaa committed:
Add Commands abstraction and AddToDoCommand

dominopizzaaaa committed:
Add Commands abstraction and AddToDoCommand

dominopizzaaaa committed:
Add AddEventCommand class

dominopizzaaaa committed:
Add AddDeadlineCommand class

dominopizzaaaa committed:
Modify AddToDoCommand class to remove bug

dominopizzaaaa committed:
Remove redundant imports

dominopizzaaaa committed:
Add ListCommand and remove isExit method in Command

dominopizzaaaa committed:
Add ExitCommand class

dominopizzaaaa committed:
Add Mark and Unmark Commands

dominopizzaaaa committed:
Add DeleteEventCommand

dominopizzaaaa committed:
Stash files temporarily

dominopizzaaaa committed:
Updated Test routes

dominopizzaaaa committed:
Updated Test routes

dominopizzaaaa committed:
Fix bugs

dominopizzaaaa committed:
Add more test cases

dominopizzaaaa committed:
Add tasks files

dominopizzaaaa committed:
Merge branch 'more-OOP'

dominopizzaaaa committed:
Organise files into packages

dominopizzaaaa committed:
Encapsulate main method

dominopizzaaaa committed:
Update

dominopizzaaaa committed:
Merge branch 'add-gradle-support'

dominopizzaaaa committed:
Add test folder

dominopizzaaaa committed:
Change properties of gradle-wrapper

dominopizzaaaa committed:
Add test case for Todo command

dominopizzaaaa committed:
Add test case for adding event task

dominopizzaaaa committed:
Touch up test cases for Todo commands and change some logic for todo, remove some redundant exceptions

dominopizzaaaa committed:
Remove useless test files

dominopizzaaaa committed:
Add test cases for all tasks adding

dominopizzaaaa committed:
Add Javadocs for commands

dominopizzaaaa committed:
Add Javadocs for exceptions and storage directories

dominopizzaaaa committed:
Add Javadocs for files

dominopizzaaaa committed:
Follow Java Coding Standard

dominopizzaaaa committed:
Add Find by keyword functionality

dominopizzaaaa committed:
Merge branch 'branch-Level-9'

dominopizzaaaa committed:
Merge branch 'branch-A-CodingStandard'

dominopizzaaaa committed:
Editted test cases

dominopizzaaaa committed:
Remove redundant files in test folder

dominopizzaaaa committed:
Change name of packages to follow Java convention

dominopizzaaaa committed:
Try out CheckStyle and make some changes

dominopizzaaaa committed:
Refactor code for addition on JavaFX

dominopizzaaaa committed:
Integrate Papadom chatbot logic with JavaFX interface

- Connected MainController to Papadom class for handling user input
- Updated MainView.fxml to reflect the chat interface
- Refactored Papadom to return responses as Strings for UI display
- Handled user input and bot responses in the JavaFX TextArea

dominopizzaaaa committed:
Debug and style interface

bmanara commented on src/main/java/papadom/commands/FindEventCommand.java:
```diff
@@ -0,0 +1,25 @@
+package papadom.commands;
+
+import papadom.exceptions.IncorrectTaskInputFormatException;
+import papadom.utils.Parser;
```
Should the import statement be at the bottom (above `import papadom.utils.ui` as it doesn't follow lexicographic ordering?

bmanara commented on src/main/java/papadom/utils/Papadom.java:
```diff
@@ -0,0 +1,93 @@
+         * @param command The string representation of the command.
+         * @return The CommandType enum corresponding to the command.
+         */
+        public static CommandType fromString(String command) {
```
I like the use of enums here! 👍

bmanara reviewed (commented):
LGTM! Just one thing to change in terms of style guide.

dominopizzaaaa committed:
Debug and Remove checkstyle when compiling

rithanisk commented on src/main/java/papadom/utils/Parser.java:
```diff
@@ -0,0 +1,88 @@
+     * @return A Deadline task.
+     * @throws IncorrectTaskInputFormatException If the input format is incorrect.
+     */
+    public Deadline deadlineTaskCreator(String details) throws IncorrectTaskInputFormatException {
```
Use verbs for method names! Can rename this method to "createDeadlineTask"

rithanisk commented on src/main/java/papadom/commands/DeleteEventCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @throws NoTaskNumberException If no task number is provided.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage)
```
Can specify the method name to describe what it is executing (ex: "executeDeleteCommand")

rithanisk commented on src/main/java/papadom/storage/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+public class TaskList {
+
+    private final Storage STORAGE;
+    private final ArrayList<Task> TASKS_LIST = new ArrayList<>();
```
Makes more sense to name this "TASK_LIST" instead of "TASKS_LIST"

rithanisk commented on src/main/java/papadom/storage/TaskList.java:
```diff
@@ -0,0 +1,151 @@
+     * @throws WrongTaskNumberException  If the task number does not exist.
+     */
+    public String unmarkTask(String text) throws NoTaskNumberException, WrongTaskNumberException {
+        String[] parts = text.split(" ");
```
Good job distinguishing between single-valued and multi-valued variables!

rithanisk reviewed (commented):
Good job overall! Just a few nits to go through and fix

dominopizzaaaa committed:
Add JAR file for JavaFX application

Add the JAR file that includes JavaFX libraries bundled to allow the
application to run on systems without JavaFX installed.

This JAR enables the application to work standalone by embedding all
necessary dependencies within the package.

Including the JAR helps streamline the deployment process, making it
easier to distribute the application without worrying about missing
JavaFX installations on the user's machine.

dominopizzaaaa committed:
Add null checks and input validation

Several methods were lacking null checks and input validation, which could lead to unexpected behavior or errors. To improve code quality and robustness, assertions have been added to ensure:
- Non-null inputs for critical methods
- Valid input formats and ranges
- Expected conditions are met before executing sensitive code

These assertions will help catch potential issues early and make the code more reliable.

dominopizzaaaa committed:
Improve code quality for clarity and maintainability

    - Removed most magic literals
    - Refactored deleteEvent, markTask, and unmarkTask methods to reduce code duplication and improve readability

dominopizzaaaa committed:
Merge pull request #2 from dominopizzaaaa/branch-A-Assertions

Add null checks and input validation

dominopizzaaaa committed:
Merge branch 'master' into branch-A-CodeQuality

dominopizzaaaa committed:
Merge pull request #1 from dominopizzaaaa/branch-A-CodeQuality

Improve code quality for clarity and maintainability

dominopizzaaaa committed:
Improve search functionality with partial keyword matching

- Enhanced search flexibility by allowing partial keyword matches in task descriptions.
- Users can now find tasks even if the keyword only partially matches the item.
- Refactored search logic for more robust and user-friendly search results.

dominopizzaaaa committed:
Merge pull request #3 from dominopizzaaaa/branch-C-BetterSearch

Improve search functionality with partial keyword matching

dominopizzaaaa committed:
Add test case for find event command and improve output formatting

dominopizzaaaa committed:
Enhance UI for chatbot interaction

- Added user instructions on how to use the bot for better usability.
- Removed extra lines before and after each bot reply to streamline the UI and create a cleaner look.

dominopizzaaaa committed:
Merge pull request #4 from dominopizzaaaa/branch-A-BetterGui

Enhance UI for chatbot interaction

dominopizzaaaa committed:
Add profile pictures for bot and user

dominopizzaaaa committed:
Update README for User Guide

dominopizzaaaa committed:
Remove minor bugs

dominopizzaaaa committed:
Arrange imports in lexicographical order

dominopizzaaaa committed:
Refactor method and variable names for better clarity and readability

dominopizzaaaa committed:
Remove unnecessary 'checkstyle' plugin reference

The 'checkstyle' plugin ID was removed as it was no longer required in the build configuration.
This change simplifies the project setup.

dominopizzaaaa committed:
Remove unused Papadom chatbot initialization from MainController

The Papadom chatbot instance was removed from the MainController as it was no longer being used.
This change cleans up unnecessary code and simplifies the class structure.

dominopizzaaaa committed:
Remove outdated run instructions from code comments

The run instructions (./gradlew clean build and ./gradlew run) were removed from the comments to make the code cleaner and neater.
This information is better suited for documentation or external guides.

dominopizzaaaa committed:
Make userImage and botImage variables final in MainController

The userImage and botImage variables were made final to indicate that they are constants and should not be reassigned, improving code safety and clarity.

dominopizzaaaa committed:
Remove unnecessary comments about Insets from MainView.fxml

Deleted the redundant comments in MainView.fxml to improve readability and reduce clutter.

dominopizzaaaa committed:
Package JavaFX libraries into fat JAR for cross-platform use

Updated build.gradle to include JavaFX dependencies in the fat JAR
using the shadow plugin. This allows the application to run on any
platform without requiring JavaFX to be installed separately.

dominopizzaaaa committed:
Fix JavaFX packaging issues in fat JAR

Updated build.gradle to include platform-specific JavaFX libraries
for Windows, macOS, and Linux. Ensured that all necessary JavaFX
modules (controls, fxml, graphics, base) are included in the fat
JAR. This resolves the issue where the JavaFX runtime components
were missing when running the application on different platforms.

dominopizzaaaa committed:
Refactor project setup and fix JavaFX initialization issues

- Updated build.gradle to include correct JavaFX dependencies
- Added Launcher class to handle Application.launch() for PapadomApp
- Removed main method from PapadomApp for JavaFX best practices
- Updated project structure to ensure correct JavaFX application launch
- Temporarily removed checkstyle for easier testing (to add back later)

dominopizzaaaa committed:
Fix file path issue for tasks.txt when running from JAR

- Updated STORAGE_PATH to use relative path instead of src/main directory
- Added logic to create necessary directories if they do not exist
- Ensured tasks.txt is created in the runtime directory or user's home

dominopizzaaaa committed:
Refactor task handling to load from .txt for session persistence

Tasks were not persisted across sessions, requiring re-entry of tasks.
Storing tasks in a .txt file ensures they are retained between sessions.

Changed the structure of task management to directly load from the
tasks.txt file and update it accordingly. This includes adding,
deleting, marking, and unmarking tasks in real-time.

dominopizzaaaa committed:
Update bot behavior and adjust test cases

- Tweaked bot's behavior to reflect new expected functionalities.
- Adjusted test files to align with updated bot responses and behavior.
- Modified related source files to implement and support these changes.

dominopizzaaaa committed:
Merge pull request #6 from dominopizzaaaa/branch-Redo-Storage

Branch redo storage

dominopizzaaaa committed:
Add Checkstyle plugin and apply style fixes

- Integrate Checkstyle plugin into build.gradle for static code analysis
- Reorder import statements to conform with CustomImportOrder rule
- Adjust variable names and definitions for AbbreviationAsWordInName rule
- Fix DeclarationOrder violations by reordering variable declarations
- Ensure single space between non-whitespace characters for consistency

dominopizzaaaa committed:
Improve method header comments

Refine the header comment for the main entry point method to provide
a more descriptive overview of its functionality. Clarified the purpose
of the args parameter while following coding standards for method comments.

dominopizzaaaa committed:
Remove redundant variables and apply minor style changes

- Removed unnecessary variables for improved code clarity.
- Applied minor style adjustments to enhance readability and adhere
  to coding standards.

dominopizzaaaa committed:
Add window close on 'bye' and listener for 'Enter'

- Close the window after a 3-second delay when the user inputs "bye",
  improving the exit experience.
- Add 'Enter' key listener for sending messages to enhance user
  interaction.
- Minor UI improvements for chat input flow and window control.
- Implement default case in key event handling for code consistency.
